src/platforms/linux-raspbian.js

```diff
@@ -131,6 +131,293 @@ function setLanMode(mode, options = {}) {
+            options.key = value;
+            break;
+          case 'wpa_pairwise':
+            if (value.indexOf('TKIP') >= 0) {
+              cipher += '+tkip';
+            }
+            if (value.indexOf('CCMP') >= 0) {
+              cipher += '+ccmp';
+            }
+            break;
+        }
+      }
+    }
+  } else {
+    mode = 'sta';
+    proc = child_process.spawnSync(
+      'wpa_cli',
+      ['-i', 'wlan0', 'status'],
+      {encoding: 'utf8'}
+    );
+
+    if (proc.status !== 0) {
+      return {enabled, mode, options};
+    }
+
+    for (const line of proc.stdout.split('\n')) {
+      const [key, value] = line.split('=', 2);
+      switch (key) {
+        case 'wpa_state':
+          enabled = line.split('=')[1] === 'COMPLETED';
+          break;
+        case 'ssid':
+          options.ssid = line.substring(5);
+          break;
+        case 'key_mgmt':
+          switch (value) {
+            case 'WPA2-PSK':
+              mgmt = 'psk2';
+              break;
+            default:
+              mgmt = value.toLowerCase();
+              break;
+          }
+          break;
+        case 'pairwise_cipher':
+          if (value.indexOf('TKIP') >= 0) {
+            cipher += '+tkip';
+          }
+          if (value.indexOf('CCMP') >= 0) {
+            cipher += '+ccmp';
+          }
+          break;
+      }
+    }
+  }
+
+  if (mgmt) {
+    options.encryption = mgmt;
+
+    if (mgmt !== 'none' && cipher) {
+      options.encryption += cipher;
+    }
+  }
+
+  return {enabled, mode, options};
+}
+
+/**
+ * Set the wireless mode and options.
+ *
+ * @param {boolean} enabled - whether or not wireless is enabled
+ * @param {string} mode - ap, sta, ...
+ * @param {Object?} options - options specific to wireless mode
+ * @returns {boolean} Boolean indicating success.
+ */
+function setWirelessMode(enabled, mode = 'ap', options = {}) {
+  const valid = ['ap', 'sta'];
+  if (enabled && !valid.includes(mode)) {
+    return false;
+  }
+
+  // First, remove existing networks
+  let proc = child_process.spawnSync(
+    'wpa_cli',
+    ['-i', 'wlan0', 'list_networks'],
+    {encoding: 'utf8'}
+  );
+  if (proc.status === 0) {
+    const networks = proc.stdout.split('\n')
+      .filter((l) => !l.startsWith('network'))
+      .map((l) => l.split(' ')[0])
+      .reverse();
+
+    for (const id of networks) {
+      proc = child_process.spawnSync(
+        'wpa_cli',
+        ['-i', 'wlan0', 'remove_network', id]
+      );
+      if (proc.status !== 0) {
+        console.log('Failed to remove network with id:', id);
+      }
+    }
+  }
+
+  // Then, disable hostapd. It will either need to be off or reconfigured, so
+  // this is valid in both modes.
+  proc = child_process.spawnSync(
+    'sudo',
+    ['systemctl', 'stop', 'hostapd.service']
+  );
+  if (proc.status !== 0) {
+    return false;
+  }
+
+  proc = child_process.spawnSync(
+    'sudo',
+    ['systemctl', 'disable', 'hostapd.service']
```

Comment:
Just a sanity check. Does this make sense here? If the system crashed/rebooted between here and when we rewrite the config file, then hostapd won't start on reboot. Just wanted to make sure that's what we actually want.

Reply:
I can move that into the `sta` check.